hpx/util/detail/basic_function.hpp

```diff
@@ -168,7 +168,7 @@ namespace hpx { namespace util { namespace detail
                     vtable::reconstruct<target_type>(&object, std::forward<F>(f));
                 } else {
                     reset();
-                    vtable::destruct<empty_function<R(Ts...)> >(&object);
+                    vtable::delete_<empty_function<R(Ts...)> >(&object);
```

Comment:
`reset()` leaves the object holding an `empty_function`, which should be guaranteed to be embedded. What's the problem reported by `clang-check` here? It might be pointing at something else going on.